src/main/java/com/wego/httpcache/services/impl/AsyncHttpCacheServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.wego.httpcache.services.impl;
+
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import com.google.inject.assistedinject.Assisted;
+import com.ning.http.client.AsyncCompletionHandlerBase;
+import com.ning.http.client.AsyncHttpClient;
+import com.ning.http.client.Request;
+import com.ning.http.client.Response;
+import com.wego.httpcache.dao.models.CachedResponse;
+import com.wego.httpcache.services.AsyncHttpCacheService;
+import com.wego.httpcache.services.CachedResponseService;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import redis.clients.util.MurmurHash;
+
+public class AsyncHttpCacheServiceImpl implements AsyncHttpCacheService {
+
+  @Inject private CachedResponseService cachedResponseService;
+
+  @Inject private AsyncHttpClient asyncHttpClient;
```

Comment:
this should not be injected since this is assisted and will be set for each provider